packages/client/src/links/httpSubscriptionLink.ts

```diff
@@ -229,12 +229,16 @@ export function unstable_httpSubscriptionLink<
               }
             }
           }
-
           observer.next({
             result: {
               type: 'stopped',
             },
           });
+          connectionState.next({
+            type: 'state',
+            state: 'idle',
```

Comment:
is idle the appropriate state for something which has completed? Would a UI need a way to easily indicate that the data stream is "finished" as opposed to not yet started?

Does Idle imply is can be easily restarted and does the userSubscription hook have a way to do that?

Reply:
Would love if you wanna do a PR to the docs to this as a follow-up -- this isn't my strong suit but I'll try to explain.

Idle just means there's no connection and no intent on creating a new one. It happens either if

- subscription is disabled
- subscription has ended on the server-side

In the case of React, you can restart the subscription with `.restart()`